tests/test_analysis/test_outliers/test_isolation_forest_outliers.py

```diff
@@ -0,0 +1,197 @@
+        ("segment_1", True, "df_segment_1"),
+        ("segment_0", False, "df_segment_0"),
+    ],
+)
+def test_prepare_segment_df(ts_with_features, segment, ignore_missing, expected_df, request):
+    expected_df = request.getfixturevalue(expected_df)
+    df = _prepare_segment_df(df=ts_with_features.to_pandas(), segment=segment, ignore_missing=ignore_missing)
+    pd.testing.assert_frame_equal(df.sort_index(axis=1), expected_df.sort_index(axis=1), check_names=False)
+
+
+def test_prepare_segment_df_fails(ts_with_features):
+    with pytest.raises(
+        ValueError,
+        match="Series segment_1 contains NaNs! Set `ignore_missing=True` to drop them or impute them appropriately!",
+    ):
+        _ = _prepare_segment_df(df=ts_with_features.to_pandas(), segment="segment_1", ignore_missing=False)
+
+
+@pytest.mark.parametrize(
+    "in_column, use_in_column, expected_anomalies",
+    [
+        ("target", True, [np.datetime64("2000-01-04")]),
+        ("target", False, []),
+        ("exog_1", True, [np.datetime64("2000-01-04")]),
+        ("exog_1", False, [np.datetime64("2000-01-04")]),
+    ],
+)
+def test_get_anomalies_isolation_forest_segment_index_only(df_segment_0, in_column, use_in_column, expected_anomalies):
+    model = IsolationForest(n_estimators=3)
+    anomalies = _get_anomalies_isolation_forest_segment(
+        df_segment=df_segment_0, model=model, in_column=in_column, use_in_column=use_in_column, index_only=True
+    )
+    assert anomalies == expected_anomalies
+
+
+@pytest.mark.parametrize(
+    "in_column,use_in_column,expected_anomalies",
+    [
+        (
+            "target",
+            True,
+            pd.Series(
+                data=[20.0],
+                index=pd.DatetimeIndex([np.datetime64("2000-01-04")], freq="D"),
+            ),
+        ),
+        (
+            "target",
+            False,
+            pd.Series(data=[], index=pd.DatetimeIndex([], freq="D"), dtype=float),
+        ),
+        (
+            "exog_1",
+            True,
+            pd.Series(
+                data=[1.0],
+                index=pd.DatetimeIndex([np.datetime64("2000-01-04")], freq="D"),
+            ),
+        ),
+        (
+            "exog_1",
+            False,
+            pd.Series(
+                data=[1.0],
+                index=pd.DatetimeIndex([np.datetime64("2000-01-04")], freq="D"),
+            ),
+        ),
+    ],
+)
+def test_get_anomalies_isolation_forest_segment_series(df_segment_0, in_column, use_in_column, expected_anomalies):
+    model = IsolationForest(n_estimators=3)
+    anomalies = _get_anomalies_isolation_forest_segment(
+        df_segment=df_segment_0, model=model, in_column=in_column, use_in_column=use_in_column, index_only=False
+    )
+    pd.testing.assert_series_equal(anomalies, expected_anomalies, check_names=False)
+
+
+def test_get_anomalies_isolation_forest_interface(ts_with_features):
+    anomalies = get_anomalies_isolation_forest(
```

Comment:
Can we add another test when one series is used to detect anomalies in the other with `index_only=False`? For example, `in_column=exog_1` and `features_to_use=['target']`

Reply:
It is already implemented in `test_get_anomalies_isolation_forest_segment_series` -- the last test case